pkg/graveler/compacted_diff_iterator.go

```diff
@@ -0,0 +1,110 @@
+package graveler
+
+import "bytes"
+
+// CompactedDiffIterator calculate the diff between commit to staged + compacted area
+type CompactedDiffIterator struct {
+	stageAndCommittedDiffIterator     DiffIterator
+	compactedAndCommittedDiffIterator DiffIterator
+	p                                 DiffIterator
+}
```

Comment:
Nothing about this is 'compacted', `stage` or `committed`. I don't think this iterator does whats stated above

Reply:
```suggestion
// UnionDiffIterator calculate the union diff between 2 iterators.
// The output iterator yields a single result for each key.
// If a key exist in the 2 iterators, iteratorA value prevails.
type JoinedDiffIterator struct {
	iterA     DiffIterator
	iterB     DiffIterator
	p           DiffIterator
}
```